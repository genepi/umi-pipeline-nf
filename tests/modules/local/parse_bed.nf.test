nextflow_process {

    name "Test Process PARSE_BED"
    script "modules/local/parse_bed.nf"
    process "PARSE_BED"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = Channel.fromPath( file("$baseDir/tests/input/parse_bed/lpa-ref2645.bed", checkIfExists: true) )
                    .splitText()
                    .map { line ->
                        def fields = line.tokenize('\t')
                        def target = fields[3].trim()
                        return tuple(target, line)
                    }
                
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should split bed into two files") {

        when {
            process {
                """
                input[0] = Channel.fromPath( file("$baseDir/tests/input/parse_bed/lpa-refcombined.bed", checkIfExists: true) )
                    .splitText()
                    .map { line ->
                        def fields = line.tokenize('\t')
                        def target = fields[3].trim()
                        return tuple(target, line)
                    }
                
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
