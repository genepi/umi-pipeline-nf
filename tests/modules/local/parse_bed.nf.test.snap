{
    "Should split bed into two files": {
        "content": [
            {
                "0": [
                    [
                        "lpa2645",
                        "lpa2645.bed:md5,0204fd1c946667aeabff838480ec9cd6"
                    ],
                    [
                        "lpa5104",
                        "lpa5104.bed:md5,c6225a429857908753c23476a58d7b98"
                    ]
                ],
                "bed_channel": [
                    [
                        "lpa2645",
                        "lpa2645.bed:md5,0204fd1c946667aeabff838480ec9cd6"
                    ],
                    [
                        "lpa5104",
                        "lpa5104.bed:md5,c6225a429857908753c23476a58d7b98"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-26T11:09:40.243912"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "lpa2645",
                        "lpa2645.bed:md5,0204fd1c946667aeabff838480ec9cd6"
                    ]
                ],
                "bed_channel": [
                    [
                        "lpa2645",
                        "lpa2645.bed:md5,0204fd1c946667aeabff838480ec9cd6"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-26T11:06:25.952017"
    }
}